sds/src/scanner/mod.rs

```diff
@@ -48,6 +48,7 @@ pub trait CompiledRuleTrait: Send + Sync {
     fn get_string_matches(
         &self,
         content: &str,
+        path: Option<&str>,
```

Comment:
As this is not the path, I think it should be named differently.
Also, there is possibly no need to need to sanitize the path on each `get_string_matches`, this will be needed only if we go in `is_false_positive_match`. Another option here could be to expose the real path in `get_string_matches` and to sanitise it only in `contains_keyword_in_path` when it's needed.

Here the advantage is that you will need to sanitise the path exactly once for each field (and not multiple time if there is multiple match). To avoid recomputing the path multiple times for a same field, you can possibly use a singleton pattern.